internal/controller/authenticator/jwtcachefiller/jwtcachefiller_test.go

```diff
@@ -363,41 +712,217 @@ func TestController(t *testing.T) {
+					unknownAuthenticatorValid(frozenMetav1Now, 0),
+					unknownJWKSURLValid(frozenMetav1Now, 0),
+				},
+			),
+			wantStatusPhase: "Error",
+		}, {
+			name: "validateProviderDiscovery: could not perform oidc discovery on provider issuer will fail sync loop and will report failed and unknown conditions and Error phase and will enqueue new sync",
+			jwtAuthenticators: []runtime.Object{
+				&auth1alpha1.JWTAuthenticator{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-name",
+					},
+					Spec: *validIssuerURLButDoesNotExistJWTAuthenticatorSpec,
+				},
+			},
+			syncKey: controllerlib.Key{Name: "test-name"},
+			wantStatusConditions: status.ReplaceConditions(
+				allHappyConditionsSuccess(goodIssuer, frozenMetav1Now, 0),
+				[]metav1.Condition{
+					happyIssuerURLValid(frozenMetav1Now, 0),
+					sadReadyCondition(frozenMetav1Now, 0),
+					sadDiscoveryURLValidConnectionRefused(someOtherLocalhostIssuer, frozenMetav1Now, 0),
+					unknownAuthenticatorValid(frozenMetav1Now, 0),
+					unknownJWKSURLValid(frozenMetav1Now, 0),
+				},
+			),
+			wantStatusPhase: "Error",
+			wantSyncLoopErr: testutil.WantExactErrorString(`could not perform oidc discovery on provider issuer: Get "` + someOtherLocalhostIssuer + `/.well-known/openid-configuration": dial tcp ` + someOtherLocalhostIssuerParsed.Host + `: connect: connection refused`),
+		},
+		// cannot be tested currently the way the coreos lib works.
+		// the constructor requires an issuer in the payload and validates the issuer matches the actual issuer,
+		// which ensures the .Claims() parsing cannot fail (in the current impl)
+		// { name: "validateProviderJWKSURL: could not get provider jwks_uri... ",},
+		{
+			name: "validateProviderJWKSURL: could not parse provider jwks_uri will fail sync loop and will report failed and unknown conditions and Error phase and will enqueue new sync",
+			jwtAuthenticators: []runtime.Object{
+				&auth1alpha1.JWTAuthenticator{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-name",
+					},
+					Spec: *badIssuerJWKSURIJWTAuthenticatorSpec,
+				},
+			},
+			syncKey: controllerlib.Key{Name: "test-name"},
+			wantStatusConditions: status.ReplaceConditions(
+				allHappyConditionsSuccess(goodIssuer, frozenMetav1Now, 0),
+				[]metav1.Condition{
+					happyIssuerURLValid(frozenMetav1Now, 0),
+					sadReadyCondition(frozenMetav1Now, 0),
+					unknownAuthenticatorValid(frozenMetav1Now, 0),
+					sadJWKSURLValidParseURI("https://.café   .com/café/café/café/coffee/jwks.json", frozenMetav1Now, 0),
+				},
+			),
+			wantStatusPhase: "Error",
+			wantSyncLoopErr: testutil.WantExactErrorString(`could not parse provider jwks_uri: parse "https://.café   .com/café/café/café/coffee/jwks.json": invalid character " " in host name`),
+		}, {
+			name: "validateProviderJWKSURL: invalid scheme, requires 'https'  will fail sync loop and will report failed and unknown conditions and Error phase and will enqueue new sync",
```

Comment:
NIT: extra space after https and before will: `invalidd scheme, requires 'https'  will`

Reply:
fixed